xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-docker/src/main/java/org/xwiki/test/docker/junit5/servletEngine/ServletContainerExecutor.java

```diff
@@ -168,6 +170,71 @@ private void startContainer(GenericContainer servletContainer)
         start(servletContainer, this.testConfiguration);
     }
 
+    private void manageDefaultServletEngine(TestConfiguration testConfiguration)
+    {
+        throw new RuntimeException(String.format("Servlet engine [%s] is not yet supported!",
+            testConfiguration.getServletEngine()));
+    }
+
+    private GenericContainer getServletContainerImage(TestConfiguration testConfiguration)
+    {
+        final String baseImageName;
+
+        switch (testConfiguration.getServletEngine()) {
+            case TOMCAT:
+                baseImageName = String.format("tomcat:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            case JETTY:
+                baseImageName = String.format("jetty:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            case WILDFLY:
+                baseImageName = String.format("jboss/wildfly:%s", testConfiguration.getServletEngineTag() != null
+                    ? testConfiguration.getServletEngineTag() : LATEST);
+                break;
+
+            default:
+                this.manageDefaultServletEngine(testConfiguration);
+                baseImageName = null;
+        }
+
+        // we want to use libreoffice so we need a custom image
+        if (testConfiguration.isOffice()) {
+            // name of the image that we will create
+            String imageName = StringUtils.join(Arrays.asList("xwiki", baseImageName, "office"), '_').replace(':', '_');
+
+            // we won't delete the image, so it's possible that the image already exists: it would avoid us to create
+            // it again
+            List<Image> imageSearchResults = DockerClientFactory.instance().client().listImagesCmd()
+                .withImageNameFilter(imageName).exec();
+
+            // the image does not exist
+            if (imageSearchResults.isEmpty()) {
+
```

Comment:
empty line not needed (that's why we put `{`on the same line, otherwise we would put it on the next line)